src/Conversion/ONNXToTorch/NN/GemmOp.cpp

```diff
@@ -0,0 +1,196 @@
+                          1, ctx) {}
+
+  LogicalResult
+  matchAndRewrite(Operation *op, ArrayRef<Value> operands,
+                  ConversionPatternRewriter &rewriter) const final {
+    ONNXGemmOp gemmOp = llvm::dyn_cast_or_null<ONNXGemmOp>(op);
+    assert(gemmOp && "Expecting op to have a strong type");
+
+    auto loc = gemmOp.getLoc();
+    mlir::MLIRContext *context = gemmOp.getContext();
+
+    auto alpha = gemmOp.alphaAttr();   // ::mlir::FloatAttr
+    auto beta = gemmOp.betaAttr();     // ::mlir::FloatAttr
+    auto transA = gemmOp.transAAttr(); // ::mlir::IntegerAttr
+    auto transB = gemmOp.transBAttr(); // ::mlir::IntegerAttr
+
+    Value iOne = getIntValue(1, rewriter, context, loc);
+
+    auto A = gemmOp.A();
+    auto B = gemmOp.B();
+    auto C = gemmOp.C();
+
+    auto aTensor = getTorchTensor(A, rewriter, context, loc);
+    auto bTensor = getTorchTensor(B, rewriter, context, loc);
+    auto cTensor = getTorchTensor(C, rewriter, context, loc);
+
+    auto aType = toTorchType(context, A.getType());
+    auto bType = toTorchType(context, B.getType());
+    auto cType = toTorchType(context, C.getType());
+    auto resultType = toTorchType(context, gemmOp.getResult().getType());
+
+    // Transpose A and B. Transpose on Torch is only 2d or less.
+    assert((getRank(A) == 2 && getRank(B) == 2 && getRank(C) <= 2) &&
+           "Checking input dimensions");
+
+    auto aShapedType = A.getType().dyn_cast<ShapedType>();
+    auto bShapedType = B.getType().dyn_cast<ShapedType>();
+
+    mlir::Type transposeAType =
+        (transA)
+            ? Torch::ValueTensorType::get(
+                  context, ArrayRef<int64_t>(getTransposedShape2D(aShapedType)),
+                  aShapedType.getElementType())
+            : aType;
+    mlir::Type transposeBType =
+        (transB)
+            ? Torch::ValueTensorType::get(
+                  context, ArrayRef<int64_t>(getTransposedShape2D(bShapedType)),
+                  bShapedType.getElementType())
+            : bType;
+
+    Value transposeAVal =
+        (transA) ? rewriter.create<AtenTOp>(loc, transposeAType, aTensor)
+                 : aTensor;
+    Value transposeBVal =
+        (transB) ? rewriter.create<AtenTOp>(loc, transposeBType, bTensor)
+                 : bTensor;
+
+    // Compute Y = alpha * A’ * B’ + beta * C
+    // Scalar multiplication with alpha(alpha * A’)
+    // and beta(beta * C) values.
+    Value alphaMulResult = NULL, betaMulResult = NULL;
+    if (alpha) {
+      Value alpha3v = getFloatValue(alpha, rewriter, loc);
+      alphaMulResult = rewriter.create<AtenMulScalarOp>(loc, transposeAType,
+                                                        transposeAVal, alpha3v);
+    }
+
+    if (beta) {
+      Value beta3v = getFloatValue(beta, rewriter, loc);
+      betaMulResult =
+          rewriter.create<AtenMulScalarOp>(loc, cType, cTensor, beta3v);
+    }
+
+    // Bmm Operation ((alpha * A’) * B’)
+    Value bmmValue;
```

Comment:
`aten.bmm` expects a tensor of rank 3. This will probably not generalize for different batch sizes either way. Going to replace with `aten.mm`. https://github.com/llvm/torch-mlir/blob/c4dcdd1e34d5cf38aa884afd95ffbc2c6000efa5/lib/Conversion/TorchToLinalg/Linear.cpp#L316